dirs/dirs.go

```diff
@@ -205,6 +275,22 @@ func StripRootDir(dir string) string {
 	return "/" + result
 }
 
+// DataHomeGlobs returns a slice of globbing expressions for the snap directories in use.
+func DataHomeGlobs(opts *SnapDirOptions) []string {
+	snapHomeDirsMu.Lock()
+	defer snapHomeDirsMu.Unlock()
+	if opts == nil {
+		opts = &SnapDirOptions{}
+	}
+	glob := make([]string, len(snapDataHomeGlob))
+	if opts.HiddenSnapDataDir {
+		copy(glob, hiddenSnapDataHomeGlob)
+		return glob
+	}
+	copy(glob, snapDataHomeGlob)
```

Comment:
I'm not sure I would worry about the copying here

Reply:
Thank you for the review! 
In this case, directly returning `snapDataHomeGlob` would cause a risk of other functions overwriting `snapDataHomeGlob`. The example below shows how a function could do so:

```
slice := dirs.DataHomeGlobs(nil)
slice[0] = "TEST"
fmt.Print(dirs.DataHomeGlobs(nil)[0]) // This will print "TEST" instead of the original value
```

This also reminds  me that I forgot to implement the slice copy for `dirs.SetSnapHomeDirs` when returning `snapHomeDirs` !

I also wrote a quick test showing this behavior:
```
package main

import "fmt"

var slice []string

func retSlice() []string {
	slice=append(slice,"hello")
	slice=append(slice,"world")
	return slice
}

func main() {
	slice2:=retSlice()
	slice2[0]="start"
	fmt.Printf("%v",slice) // will print [start world] instead of [hello world]
}

```

Reply:
I understand, I just don't think modifying the result of those functions is a likely pattern in our code